inbox/mailsync/service.py

```diff
@@ -349,14 +368,8 @@ def start_sync(self, account_id: int) -> bool:
                 return False
         return True
 
-    def stop(self, *args):
-        self.log.info("stopping mail sync process")
-        for _, v in self.email_sync_monitors.items():
-            gevent.kill(v)
-        for _, v in self.contact_sync_monitors.items():
-            gevent.kill(v)
-        for _, v in self.event_sync_monitors.items():
-            gevent.kill(v)
```

Comment:
Moved this entire block to `run()` since you should not be killing greenlets/threads from a signal handler. You should just set a flag which we do, and then once the loop breaks in `run()` because `keep_running` becomes `False` we do it outside signal handler.